docs/infrastructure/logging.md

```diff
@@ -257,6 +264,116 @@ determines how often logger should send its collected events to sentry.io
+in
+[#12464](https://github.com/enso-org/enso/issues/12464#issuecomment-2713545076).
+
+### Logger namespace
+
+To send telemetry data to our cloud endpoint, a logger inside
+`org.enso.telemetry` namespace must be used. To create such a logger from Java,
+use something like this:
+
+```java
+var logger = org.slf4j.LoggerFactory.getLogger("org.enso.telemetry.MyLogger");
+```
+
+To create this logger in Enso, use:
+
+```
+polyglot java import org.slf4j.LoggerFactory
+logger = LoggerFactory.getLogger "org.enso.telemetry.MyLogger"
+```
+
+### LogEvent format
+
+The format of the message passed to the logger is important. If it does not
+follow this specification, it will not be forwarded to the cloud endpoint. The
+message consist of two parts: `message` and `arguments` delimited by a colon:
+`<message>: <arguments>`. The `message` part is a string that describes the
+event. The `arguments` part is a mapping of `name={}` pairs separated by a
+comma. Note that for the compatibility with standard `ch.qos.logback` appenders,
+it is important that the value of argument is `{}`. The actual argument values
+are passed to the message as argument array via
+`ILoggingEvent.getArgumentArray()` method. Number of these arguments must match
+number of the arguments given in the message.
+
+An example in Java is:
+
+```java
+logger.trace("MyEvent: name={}, age={}", "John", 42);
+```
+
+In Enso, one has to use
+[org.slf4j.spi.LoggingEventBuilder](https://github.com/qos-ch/slf4j/blob/master/slf4j-api/src/main/java/org/slf4j/spi/LoggingEventBuilder.java)
+like this:
+
+```
+event_bldr = logger.atTrace
+event_bldr.setMessage "MyEvent: name={}, age={}"
+event_bldr.addArgument "John"
+event_bldr.addArgument 42
+event_bldr.log
+```
+
+The reason for that is that the `trace` method accepts vararg and there is no
+way to pass a vararg arguments from Enso to Java.
+
+### Transforming LogEvent to HTTP POST request
+
+If the LogEvent has the correct aforementioned format, it is delegated to the
+`TelemetryAppender` that is responsible for sending HTTP POST requests
+asynchronously to the cloud endpoint. The payload of the message looks roughly
+like this:
+
+```json
+{
+  "message": "MyEvent",
+  "metadata": {
+    "loggerName": "org.enso.telemetry.MyLogger",
+    "name": "John",
+    "age": 42
+  }
+}
+```
+
+The actual format expected by the cloud is defined at
+[logs/endpoints/remote.rs](https://github.com/enso-org/cloud-v2/blob/main/src/lambdas/src/lambdas/logs/endpoints/remote.rs#L169).
+
+If the format of the LogEvent does not follow the specification, there will be a
+warning in the logs, and nothing will be sent.
+
+### Log level
+
+TelemetryAppender is enabled by default for all logging levels. If you wish to
+send telemetry event only to the TelemetryAppender, use `TRACE` level. If you
+use `DEBUG` level, it will, by default, be also send to the FileAppender. If you
+use `INFO` level, it will, by default, be also send to the FileAppender and
+ConsoleAppender.
```

Comment:
Out of curiosity: 
- can I change log level of just `org.enso.telemetry` namespace?
- without changing the code (e.g. `application.conf` files)?
- how can I do it for CLI?

Reply:
Currently, Telemetry Appender cannot be disabled - explained in 791cee5e91c183b2aed7fa2fece1baba65d9dd16. This is a limitation now that can be alleviated in the future, if we want to be able to disable it.

Reply:
No telemetry is collected when running via CLI - cd0eff9cf710332cd226031cfaec6ba8ce9ffb47